hooks/useCurrency.ts

```diff
@@ -0,0 +1,33 @@
+import { useState, useEffect, useCallback } from 'react';
+import { addCurrencyAmount, getCurrencyAmount, reduceCurrencyAmount } from '@/database/queries';
+
+const useCurrency = () => {
+    const [currency, setCurrency] = useState<number>(0);
+    const [refresh, setRefresh] = useState<number>(0);
+
+    const fetchCurrency = useCallback(async () => {
+        let fetchedCurrency = await getCurrencyAmount();
+        setCurrency(fetchedCurrency);
+    }, []);
+
+    useEffect(() => {
+        fetchCurrency();
+    }, [fetchCurrency, refresh]);
+
+    const refreshCurrency = () => {
+        setRefresh(prev => prev + 1);
+    };
+
+    const addCurrency = async (amount: number) => {
+        await addCurrencyAmount(amount);
+        refreshCurrency();
+
+    }
+    const reduceCurrency = async (amount: number) => {
+        await reduceCurrencyAmount(amount);
+        refreshCurrency();
+    }
+    return { currency, addCurrency, reduceCurrency, refreshCurrency };
+};
+
+export default useCurrency;
```

Comment:
My understanding is that the goal is to provide currency state to different components. The best way to do that is to create a context. (Though see my comments about if currency is needed)